### ISSUE TITLE
configuration file permission

### ISSUE THREAD

**Issue by faxm0dem** [balabit/syslog-ng#1106]
Consider the following shell snippet:

```
☠ id
uid=1234(fabien) gid=1234(wernli)
☠ touch /tmp/foo && chmod 600 /tmp/foo && sudo syslog-ng -f /tmp/foo -s
```

This fails on RHEL6:

```
[2016-06-29T14:38:05.865105] Error opening configuration file; filename='/tmp/foo', error='Permission denied (13)'
```

But not on RHEL7:

```
[2016-06-29T14:38:37.732071] WARNING: Configuration file has no version number, assuming syslog-ng 2.1 format. Please add @version: maj.min to the beginning of the file to indicate this explicitly;
```

SELinux is disabled. Any idea what escapes my reasoning here?

**Comment by bazsi** [comment]
We drop unneeded posix capabilities after starting up, among others
CAP_DAC_OVERRIDE which allows root to read files not in his possession.

You can tell syslog-ng not to do this (--no-caps) or change the ownership
of the file to the user running syslog-ng (e..g. root)

**Comment by faxm0dem** [comment]
spot on! on one of the systems it was disabled! @czanik any spécifications reason you disabled it on your copr packages for el7 (but not el6)?

**Comment by faxm0dem** [comment]
spot on! on one of the systems it was disabled! @czanik any specific reason you disabled it on your copr packages for el7 (but not el6)?

**Comment by czanik** [comment]
I did not change anything in the related spec file:
* ./configure uses: --with-linux-caps=auto
* it also has BuildRequires: libcap-devel

Still, it seems to be disabled:
```
[root@czprhel72 ~]# syslog-ng -V
syslog-ng 3.7.3
Installer-Version: 3.7.3
Revision: 
Compile-Date: Mar 24 2016 10:55:14
Available-Modules: afamqp,affile,afprog,afsocket,afstomp,afuser,basicfuncs,confgen,cryptofuncs,csvparser,dbparser,graphite,json-plugin,kvformat,linux-kmsg-format,pseudofile,sdjournal,syslogformat,system-source
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: off
[root@czprhel72 ~]# 
```

**Comment by faxm0dem** [comment]
as I Saïd it's enabled on your el6 package so I guess libcap-dev was missing on your build env

**Comment by czanik** [comment]
No, it's there. That's why I use ''mock'' with standardized build chroots. And according to build logs, capabilities support is found: https://copr-be.cloud.fedoraproject.org/results/czanik/syslog-ng37/epel-7-x86_64/00170854-syslog-ng/build.log.gz

Looking at the binary also shows, that libcap is linked:
```
[root@czprhel72 ~]# ldd /usr/sbin/syslog-ng
	linux-vdso.so.1 =>  (0x00007ffdd8f95000)
	libsyslog-ng-3.7.so.0 => /lib64/libsyslog-ng-3.7.so.0 (0x00007fb9a6dad000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fb9a6ba5000)
	libnsl.so.1 => /lib64/libnsl.so.1 (0x00007fb9a698b000)
	libgmodule-2.0.so.0 => /lib64/libgmodule-2.0.so.0 (0x00007fb9a6787000)
	libgthread-2.0.so.0 => /lib64/libgthread-2.0.so.0 (0x00007fb9a6585000)
	libglib-2.0.so.0 => /lib64/libglib-2.0.so.0 (0x00007fb9a624d000)
	libevtlog.so.0 => /lib64/libevtlog.so.0 (0x00007fb9a6048000)
	libcap.so.2 => /lib64/libcap.so.2 (0x00007fb9a5e43000)
	libpcre.so.1 => /lib64/libpcre.so.1 (0x00007fb9a5be1000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fb9a59dd000)
[...]
```

**Comment by lbudai** [comment]
@faxm0dem , @czanik :  any update on this?

**Comment by czanik** [comment]
Nothing I can update: according to logs capabilities related dependencies are found during configure and enabled during compilation, the library is even linked to syslog-ng, still syslog-ng reports it as "off".

I'm happy to provide a test environment on my laptop, once I'm in the office (next Monday).

**Comment by lbudai** [comment]
@czanik :  I hope we are more professional :-) We should have to be able to solve this without your laptop. You  could provide us a docker image (or any other format you prefer) with the same environment.

**Comment by MrAnno** [comment]
I don't have details on how, but this issue has been fixed in v3.9.1-2 (czanik-syslog-ng39/packages/syslog-ng-3.9.1-2.el7).

The latest versions (3.16.1-1.el7) is also built correctly (`Enable-Linux-Caps: on`), so I close this issue.

For more information, please ask @czanik.
